### PR TITLE
Separate iOS specific code (Fix tvOS build)

### PR DESCRIPTION
**Pull request by Krisztiaan**
Closes #493 with changes applied and the original author @hufkens credited. @WoLewicki let me know if this would be ok to merge, as we would be using screens on tvOS as well, and currently need `patch-package` for this. Additional flags by @gaborkajtar.

### CODE REVIEW

**Comment by WoLewicki** [comment]
So you checked and these are all places where this should be applied so it compiles on tvOS?

**Comment by krisztiaan-uic** [comment]
This way it compiles for tvOS, however, we're still in active initial development of the apps, and are on the lookout if any other modifications are necessary. Currently using these exact changes via `patch-package`.

File: ios/RNSScreen.m [hunk 1]
```diff
@@ -88,10 +88,12 @@ - (void)setStackPresentation:(RNSScreenStackPresentation)stackPresentation
       break;
     case RNSScreenStackPresentationFullScreenModal:
       _controller.modalPresentationStyle = UIModalPresentationFullScreen;
+#if (TARGET_OS_IOS)
       break;
```
Comment (WoLewicki) [inline ios/RNSScreen.m:92]:
```suggestion
      break;
#if (TARGET_OS_IOS)
```

File: ios/RNSScreenStackHeaderConfig.m [hunk 2]
```diff
@@ -383,6 +388,8 @@ + (void)updateViewController:(UIViewController *)vc withConfig:(RNSScreenStackHe
     }
     navitem.largeTitleDisplayMode = config.largeTitle ? UINavigationItemLargeTitleDisplayModeAlways : UINavigationItemLargeTitleDisplayModeNever;
   }
+#endif
+
 #ifdef __IPHONE_13_0
```
Comment (WoLewicki) [inline ios/RNSScreenStackHeaderConfig.m:393]: Why there is no `#if defined(__IPHONE_13_0) && TARGET_OS_IOS` but it is like that in line 537?

Comment (Krisztiaan) [reply ios/RNSScreenStackHeaderConfig.m]: This is likely an oversight, changed all other occurences to this.

File: ios/RNSScreen.m [hunk 3]
```diff
@@ -413,4 +417,3 @@ @implementation RCTConvert (RNSScreen)
 
 
 @end
-
```
Comment (WoLewicki) [inline ios/RNSScreen.m:416]: There should be blank line at the end of the file.

**Comment by WoLewicki** [comment]
Thank you for your contribution! Looks like it is ready to be merged :tada:

**Comment by WoLewicki** [comment]
Most probably next week.

**Pull request merged**